src/Sulu/Bundle/RouteBundle/Content/Type/PageTreeRouteContentType.php

```diff
@@ -0,0 +1,243 @@
+
+    /**
+     * @var DocumentManagerInterface
+     */
+    private $documentManager;
+
+    /**
+     * @var DocumentRegistry
+     */
+    private $documentRegistry;
+
+    /**
+     * @var ChainRouteGeneratorInterface
+     */
+    private $chainRouteGenerator;
+
+    /**
+     * @var ConflictResolverInterface
+     */
+    private $conflictResolver;
+
+    /**
+     * @var EntityManagerInterface
+     */
+    private $entityManager;
+
+    public function __construct(
+        DocumentManagerInterface $documentManager,
+        DocumentRegistry $documentRegistry,
+        ChainRouteGeneratorInterface $chainRouteGenerator,
+        ConflictResolverInterface $conflictResolver,
+        EntityManagerInterface $entityManager
+    ) {
+        parent::__construct('PageTreeRoute');
+
+        $this->documentManager = $documentManager;
+        $this->documentRegistry = $documentRegistry;
+        $this->chainRouteGenerator = $chainRouteGenerator;
+        $this->conflictResolver = $conflictResolver;
+        $this->entityManager = $entityManager;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function read(NodeInterface $node, PropertyInterface $property, $webspaceKey, $languageCode, $segmentKey)
+    {
+        $propertyName = $property->getName();
+
+        $value = [
+            'page' => $this->readPage($propertyName, $node),
+            'path' => $node->getPropertyValueWithDefault($propertyName, ''),
+            'suffix' => $node->getPropertyValueWithDefault($propertyName . '-suffix', null),
+        ];
+
+        $property->setValue($value);
+
+        return $value;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function write(
+        NodeInterface $node,
+        PropertyInterface $property,
+        $userId,
+        $webspaceKey,
+        $languageCode,
+        $segmentKey
+    ) {
+        $value = $property->getValue();
+        if (!$value) {
+            $this->remove($node, $property, $webspaceKey, $languageCode, $segmentKey);
+
+            $value = [
+                'page' => ['uuid' => null, 'path' => '/'],
+                'suffix' => null,
+            ];
+        }
+
+        $page = $this->getAttribute('page', $value, ['uuid' => null, 'path' => '/']);
+
+        $suffix = $this->getAttribute('suffix', $value);
+        if (!$suffix) {
+            $suffix = $this->generateSuffix($node, $languageCode);
+        } else {
+            $suffix = trim($suffix, '/');
+        }
+
+        $path = rtrim($page['path'], '/') . '/' . $suffix;
+        $path = $this->resolveConflicts($path);
+        $suffix = '/' . $this->getSuffix($path, $page['path'] ?? '/');
+
+        $propertyName = $property->getName();
+        $node->setProperty($propertyName, $path);
+        $node->setProperty($propertyName . '-suffix', $suffix);
+
+        $pagePropertyName = $propertyName . '-page';
+        if ($node->hasProperty($pagePropertyName)) {
+            $node->getProperty($pagePropertyName)->remove();
```

Comment:
Can you at least try to explain what's happening here? :thinking:

Reply:
I don't know the purpose of this code. The result of this code is to remove the whole property instead of just setting it to null, if the page has been set to null in the admin. Maybe @wachterjohannes can help here.